cmd/flakechecker/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	_ "embed"
+	"flag"
+	"fmt"
+	"github.com/carlmjohnson/versioninfo"
+	junit "github.com/joshdk/go-junit"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/stackrox/junit2jira/pkg/testcase"
+	"os"
+)
+
+const totalRunsLimit = 30
+
+const errDescAboveThreshold = "allowed flake ratio for test is above threshold"
+const errDescGetRatio = "get ratio for test failed"
+const errDescNoFailedTests = "no failed tests to process"
+const errDescNoMatch = "there is no match in allowed flake tests"
+const errDescShortHistory = "total runs for test is under history count threshold"
+
+type flakeCheckerParams struct {
+	junitReportsDir string
+	configFile      string
+
+	jobName      string
+	orchestrator string
+
+	dryRun bool
+}
+
+func main() {
+	var debug bool
+	var err error
+
+	p := flakeCheckerParams{}
+	flag.StringVar(&p.junitReportsDir, "junit-reports-dir", os.Getenv("ARTIFACT_DIR"), "Dir that contains jUnit reports XML files")
+	flag.StringVar(&p.configFile, "config-file", "", "Config file with defined failure ratios")
+
+	flag.StringVar(&p.jobName, "job-name", "", "Name of CI job.")
+	flag.StringVar(&p.orchestrator, "orchestrator", "", "orchestrator name (such as GKE or OpenShift), if any.")
+
+	flag.BoolVar(&p.dryRun, "dry-run", false, "When set to true issues will NOT be created.")
+	flag.BoolVar(&debug, "debug", false, "Enable debug log level")
+	versioninfo.AddFlag(flag.CommandLine)
+	flag.Parse()
+
+	if debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	err = p.run()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type recentFlakyTestInfo struct {
+	JobName   string
+	Name      string
+	Classname string
+	TotalAll  int
+	FailRatio int
+}
+
+func (p *flakeCheckerParams) checkFailedTests(bqClient biqQueryClient, failedTests []testcase.TestCase, flakeCheckerRecs []*flakeDetectionPolicy) error {
+	if len(failedTests) == 0 {
+		return errors.New(errDescNoFailedTests)
+	}
+
+	for _, failedTest := range failedTests {
+		found := false
+		log.Infof("Checking failed test: %q / %q / %q", p.jobName, failedTest.Name, failedTest.Classname)
+		for _, flakeCheckerRec := range flakeCheckerRecs {
+			match, err := flakeCheckerRec.matchJobName(p.jobName)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			match, err = flakeCheckerRec.matchTestName(failedTest.Name)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			match, err = flakeCheckerRec.matchClassname(failedTest.Classname)
+			if err != nil {
+				return err
+			}
+
```

Comment:
Please extract at least this piece to a function/method so that `checkFailedTests` fits on the screen.